gobblin-compaction/src/main/java/gobblin/compaction/mapreduce/MRCompactorJobRunner.java

```diff
@@ -499,8 +499,24 @@ private void deleteAdditionalInputPaths() throws IOException {
   }
 
   private void submitSlaEvent(Job job) {
-    CompactionSlaEventHelper.populateState(this.dataset, Optional.of(job), this.fs);
-    new SlaEventSubmitter(this.eventSubmitter, "CompactionCompleted", this.dataset.jobProps().getProperties()).submit();
+    try {
+      CompactionSlaEventHelper
+          .getEventSubmitterBuilder(this.dataset, Optional.of(job), this.fs)
+          .eventSubmitter(this.eventSubmitter)
+          .eventName(CompactionSlaEventHelper.COMPACTION_COMPLETED_EVENT_NAME)
+          .additionalMetadata(
+              CompactionRecordCountEvent.LATE_RECORD_COUNT,
+              Long.toString(this.lateOutputRecordCountProvider.getRecordCount(this.getApplicableFilePaths(this.dataset
```

Comment:
`getApplicableFilePaths` make recursive `listFileStatus` calls, which can be expensive. Is that possible to cache the late record count for sending the event?